document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/data/accounts/AccountsManager.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.companieshouse.document.generator.accounts.data.accounts;
+
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.companieshouse.api.model.accounts.Accounts;
+import uk.gov.companieshouse.environment.impl.EnvironmentReaderImpl;
+
+public class AccountsManager {
+
+    /** represents the Authorization header name in the request */
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+
+    private static final RestTemplate restTemplate = createRestTemplate();
+
+    /**
+     * Try get transaction if exists
+     *
+     * @param link - self link for the accounts object
+     * @return transaction object along with the status or not found status.
+     */
+    public static ResponseEntity<Accounts> getAccounts(String link) {
+        HttpHeaders requestHeaders = new HttpHeaders();
+        requestHeaders.set(AUTHORIZATION_HEADER, getApiKey());
+
+        HttpEntity requestEntity = new HttpEntity(requestHeaders);
+        return getAccounts(link, requestEntity);
+    }
+
+    /**
+     * Creates the rest template when class first loads
+     *
+     * @return Returns a statically created rest template
+     */
+    private static RestTemplate createRestTemplate() {
+        HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
+        return new RestTemplate(requestFactory);
+    }
+
+
+
+    /**
+     * GET the transaction resource
+     *
+     * @param url - url to send the get request
+     * @param requestEntity - the request entity object
+     */
+    private static ResponseEntity<Accounts> getAccounts(String url, HttpEntity requestEntity) {
+        return restTemplate.exchange(getRootUri() + url, HttpMethod.GET, requestEntity, Accounts.class);
+    }
+
+    /**
+     * Get the root uri from the properties file
+     *
+     * @return Get the root uri if set, otherwise throw an exception
+     */
+    private static String getRootUri() {
+        return new EnvironmentReaderImpl().getMandatoryString("API_URL");
```

Comment:
If a static `EnvironmentReaderImpl` is instantiated at the class level, I believe we can have static Strings on the class to hold these details; that way, Strings are initialised once and we don't create a new reader instance in every method call. Something like:

```
public class Class {

   private static final EnvironmentReader READER = new EnvironmentReaderImpl();

   private static String API_URL = READER.getMandatoryString("API_URL");

}
```

The Strings might even be able to be made final - not sure about that without checking though.

The added benefit of this is that any `MissingEnvironmentVariableException`'s will be thrown on class instantiation, as opposed to when the existing method is called.

Reply:
(I know this will be stripped out, but it can't hurt to do it in an efficient way for the time being.)

Reply:
Its worth noting that this will be replaced by the `private-sdk' soon so we shouldn't get too tied down to making this perfect - however you raise a good point so I will change.